backup

```diff
@@ -54,10 +66,21 @@ restic --repo="${RESTIC_REPOSITORY}" backup "${backup_args[@]}" "${tag_options[@
 rc=$?
 set -e
 
+success="true"
 if [ $rc -ne 0 ]; then
-  if [ $rc -eq 3 ] && [ -n "${POST_COMMANDS_INCOMPLETE:-}" ]; then
-  	run_commands "${POST_COMMANDS_INCOMPLETE:-}"
+  if [ $rc -eq 3 ]; then
+    if [ "$(check_bool "${SUCCESS_ON_INCOMPLETE_BACKUP:-}")" = "true" ]; then
+      run_commands "${POST_COMMANDS_INCOMPLETE:-}"
+    else
+      success="false"
+      if [ -n "${POST_COMMANDS_INCOMPLETE:-}" ]; then
+        run_commands "${POST_COMMANDS_INCOMPLETE:-}"
```

Comment:
Shouldn't you be running `_INCOMPLETE` and `_FAILURE` commands in this scenario, always?

Reply:
No, the main point of this PR is to _not_ trigger the "fatal failure" commands if the backup is only incomplete.

But if the user did not set `POST_COMMANDS_INCOMPLETE`, the behaviour should be as before (running `POST_COMMANDS_FAILURE`), in order to be able to preserve the old behaviour and to be backwards compatible.